src/apprunner/commands/deleteService.ts

```diff
@@ -20,31 +21,23 @@ function validateName(name: string) {
 }
 
 export async function deleteService(node: AppRunnerServiceNode): Promise<void> {
-    let telemetryResult: telemetry.Result = 'Failed'
     const appRunnerServiceStatus = node.info.Status as telemetry.AppRunnerServiceStatus
 
-    try {
-        const inputBox = createInputBox({
-            title: localize('AWS.apprunner.deleteService.title', 'Delete App Runner service'),
-            placeholder: localize('AWS.apprunner.deleteService.placeholder', 'delete'),
-            buttons: [createHelpButton('https://docs.aws.amazon.com/apprunner/latest/dg/manage-delete.html')],
-            validateInput: validateName,
-        })
-
-        const userInput = await inputBox.prompt()
-
-        if (!isValidResponse(userInput)) {
-            telemetryResult = 'Cancelled'
-            return
-        }
-
-        await node.delete()
-        telemetryResult = 'Succeeded'
-    } finally {
-        telemetry.recordApprunnerDeleteService({
-            result: telemetryResult,
-            appRunnerServiceStatus,
-            passive: false,
-        })
+    getTelemetryLogger('ApprunnerDeleteService').recordAppRunnerServiceStatus(appRunnerServiceStatus)
+
+    const inputBox = createInputBox({
+        title: localize('AWS.apprunner.deleteService.title', 'Delete App Runner service'),
+        placeholder: localize('AWS.apprunner.deleteService.placeholder', 'delete'),
+        buttons: [createHelpButton('https://docs.aws.amazon.com/apprunner/latest/dg/manage-delete.html')],
+        validateInput: validateName,
+    })
+
+    const userInput = await inputBox.prompt()
+
+    if (!isValidResponse(userInput)) {
+        getTelemetryLogger('ApprunnerDeleteService').recordResult('Cancelled')
```

Comment:
Because 'Cancelled' is very common we should just throw a specific error and let the core code figure it out.